### PR TITLE
Feedback box js

### PR DESCRIPTION
**Pull request by Danwhy**
### To be merged after #250 
ref #191 - Submits resource feedback box by javascript - doesn't refresh page

### CODE REVIEW

File: web/controllers/likes_controller.ex [hunk 1]
```diff
@@ -1,19 +1,32 @@
...
+          result: render_to_string(
+            App.HomepageView,
+            "resource.html",
+            resource: Resources.get_single_resource(conn, id), conn: conn
+          ),
+          id: id
+        })
+      _ ->
+        redirect(conn, to: redirect_path(conn, query))
+    end
```
Comment (shouston3) [inline web/controllers/likes_controller.ex]: At some point we should look at drying out the repetition between this function

File: web/controllers/spreadsheet_controller.ex [hunk 2]
```diff
@@ -82,4 +83,20 @@ defmodule App.SpreadsheetController do
...
+          result: render_to_string(
+            App.HomepageView,
+            "resource.html",
+            resource: Resources.get_single_resource(conn, id), conn: conn
+          ),
+          id: id
+        })
+      _ ->
+        redirect(conn, to: homepage_path(conn, :index))
+    end
```
Comment (shouston3) [inline web/controllers/spreadsheet_controller.ex]: And this function

**Review by samhstn — APPROVED** [review]
Good for now. Let's get this in for the demo tomorrow

File: web/router.ex [hunk 3]
```diff
@@ -2,7 +2,7 @@ defmodule App.Router do
   use App.Web, :router
 
   pipeline :browser do
-    plug :accepts, ["html"]
+    plug :accepts, ["html", "json"]
```
Comment (SimonLab) [inline web/router.ex]: As discussed yesterday, maybe we could use the api pipeline for ajax request

File: web/static/js/like.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
+if (isNotIE8()) {
+  module.exports = (function() {
+    selectAll(".like-form", ".dislike-form", ".resource-feedback").forEach(function(el) {
+      el.addEventListener("submit", formListener);
+    });
+  })();
+
+  function formListener(e) {
+      e.preventDefault();
+      console.log(new FormData(e.target));
```
Comment (SimonLab) [inline web/static/js/like.js]: do you want to keep this console.log?

File: web/static/js/like.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+      makePhoenixFormRequest("POST", e.target, function(err, res) {
+        var response = JSON.parse(res);
+        var resource = select("#resource_" + response.id);
+        console.log(response);
```
Comment (SimonLab) [inline web/static/js/like.js]: and this one?

File: web/static/js/vendor/helpers.js [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+// https://www.gov.uk/service-manual/technology/designing-for-different-browsers-and-devices
+function isNotIE8() {
+  return Element.prototype.addEventListener && Array.prototype.forEach;
+}
+
+// Make an AJAX request with a Phoenix form using the csrf value of that form
+function makePhoenixFormRequest(type, form, callback) {
+  var request = new XMLHttpRequest();
+  var url = form.action;
+  var csrf = form.children["_csrf_token"].value;
```
Comment (SimonLab) [inline web/static/js/vendor/helpers.js]: if the api pipleline is used for ajax request I think we can avoid adding the crsf token

**Review by SimonLab — COMMENTED** [review]
@Danwhy just two console.logs, do you want to keep them or can we delete them?
Otherwise as discussed yesterday maybe we can use the api pipeline for ajax request but not a high priority at the moment

**Comment by Danwhy** [comment]
@SimonLab I'll delete those console logs now. I will look into getting the api pipeline working later

**Comment by SimonLab** [comment]
@Danwhy sorry a merge conflict appears :worried:

**Review by SimonLab — APPROVED** [review]
:+1:

**Comment by SimonLab** [comment]
Thanks @Danwhy code looks good. It seems that codecov as some issues to retrieve previous report but I don't think it's like to the PR

**Pull request merged**